src/lib/loaders/loaders_without_authentication/gravity.ts

```diff
@@ -247,6 +247,8 @@ export default (opts) => {
     saleLoader: batchSaleLoader,
     salesLoader: batchSalesLoader,
     salesLoaderWithHeaders: gravityLoader("sales", {}, { headers: true }),
+    saleAgreementLoader: gravityLoader((id) => `sale_agreements/${id}`),
+    saleSaleAgreementLoader: gravityLoader((id) => `sale/${id}/sale_agreement`),
```

Comment:
This loader is In both gravity files (with and without authentication) since we need to return different responses depending on whether or not a user is authorized to read or manage certain sale agreements. For example, `content_manager`s can read and manage `archived` sale agreements. 

If I'm misunderstanding the best way to support that kind of flow, lmk.

Reply:
MP doesn't know if a user can read/manage sale agreements, it just knows if an incoming request has an `X-Access-Token` header included or not. If so, it uses the 'with' loaders, which propagate that header to Gravity (which will loudly fail). If not, it uses these loaders (which means they can be cached).

In general, you thus only need something in the 'without' file if you intend it to be used by logged-out users. Otherwise, you could always just use the 'with'.

In this case, it does sound like retrieving the sale agreement for a particular sale is something that's required for logged-out users (so they can see it on the sale page). However, I'm not quite clear (based on data modeling), why two data loaders (`saleSaleAgreement` and `saleAgreement`) are needed for that.

Reply:
> owever, I'm not quite clear (based on data modeling), why two data loaders (saleSaleAgreement and saleAgreement) are needed for that.

We should be able to remove `saleSaleAgreement` with the [addition](https://github.com/artsy/gravity/pull/17540) of the `bridged_has_one` helper in `gravity`.